make-release.sh

```diff
@@ -0,0 +1,146 @@
+      git branch "${PR_BRANCH}"	
+      git checkout "${PR_BRANCH}"	
+      git pull origin "${PR_BRANCH}"	
+      git push origin "${PR_BRANCH}"	
+      lastCommitComment="$(git log -1 --pretty=%B)"	
+      hub pull-request -f -m "${lastCommitComment}" -b "${BUMP_BRANCH}" -h "${PR_BRANCH}"	
+    fi 	
+  fi	
+  git checkout "${CURRENT_BRANCH}"	
+}	
+
+usage ()	
+{	
+  echo "Usage: $0 --version [VERSION TO RELEASE]"	
+  echo "Example: $0 --version 0.1.0"; echo	
+}	
+
+if [[ ! ${VERSION} ]]; then	
+  usage	
+  exit 1	
+fi	
+
+
+# derive branch from version	
+BRANCH=${VERSION%.*}.x	
+
+# if doing a .0 release, use main branch; if doing a .z release, use $BRANCH	
+if [[ ${VERSION} == *".0" ]]; then	
+  BASEBRANCH="${MAIN_BRANCH}"	
+else 	
+  BASEBRANCH="${BRANCH}"	
+fi	
+
+# work in tmp dir
+if [[ $TMP ]] && [[ -d $TMP ]]; then
+  pushd "$TMP" > /dev/null || exit 1
+  # get sources from ${BASEBRANCH} branch
+  echo "Check out ${REPO} to ${TMP}/${REPO##*/}"
+  git clone "${REPO}" -q
+  cd "${REPO##*/}" || exit 1
+fi
+
+
+# get sources from ${BASEBRANCH} branch	
+git fetch origin "${BASEBRANCH}":"${BASEBRANCH}" || true
+git checkout "${BASEBRANCH}"	
+
+# create new branch off ${BASEBRANCH} (or check out latest commits if branch already exists), then push to origin
+if [[ "${BASEBRANCH}" != "${BRANCH}" ]]; then
+  git branch "${BRANCH}" || git checkout "${BRANCH}" && git pull origin "${BRANCH}" || true
+  git push origin "${BRANCH}"
+  git fetch origin "${BRANCH}:${BRANCH}" || true
+  git checkout "${BRANCH}"
+else
+  git fetch origin "${BRANCH}:${BRANCH}" || true
+  git checkout ${BRANCH}
+fi
+set -e
+
+# change VERSION file	
+echo "${VERSION}" > VERSION
+
+QUAY_REPO="quay.io/devfile/devworkspace-controller:${VERSION}"
+docker build -t "${QUAY_REPO}" -f ./build/Dockerfile .
+docker push "${QUAY_REPO}"
+
+# replace image version in default environment
+sed -i "s/IMG=quay.io\/devfile\/devworkspace-controller:.*/IMG=quay.io\/devfile\/devworkspace-controller:${VERSION}/" ./deploy/generate-deployment.sh
+
+set -x
+bash -x ./deploy/generate-deployment.sh --use-defaults
+# tag the release	
+git tag "${VERSION}"	
+git push origin "${VERSION}"	
+COMMIT_MSG="[release] Release ${VERSION}"	
+git commit -asm "${COMMIT_MSG}"	
+
+# now update ${BASEBRANCH} to the new snapshot version	
+git checkout "${BASEBRANCH}"	
+
+# change VERSION file + commit change into ${BASEBRANCH} branch	
+if [[ "${BASEBRANCH}" != "${BRANCH}" ]]; then	
+  # bump the y digit, if it is a major release	
+  [[ $BRANCH =~ ^([0-9]+)\.([0-9]+)\.x ]] && BASE=${BASH_REMATCH[1]}; NEXT=${BASH_REMATCH[2]}; (( NEXT=NEXT+1 )) # for BRANCH=7.10.x, get BASE=7, NEXT=11	
+  NEXT_VERSION_Y="${BASE}.${NEXT}.0-SNAPSHOT"	
+  bump_version "${NEXT_VERSION_Y}" "${BASEBRANCH}"	
+fi	
+# bump the z digit	
+[[ $VERSION =~ ^([0-9]+)\.([0-9]+)\.([0-9]+) ]] && BASE="${BASH_REMATCH[1]}.${BASH_REMATCH[2]}"; NEXT="${BASH_REMATCH[3]}"; (( NEXT=NEXT+1 )) # for VERSION=7.7.1, get BASE=7.7, NEXT=2	
```

Comment:
just minor: maybe better to have example of version with VERSION scheme of devworkspace and not che
like 0.1.1 / 0.1

Reply:
ok